gcp/modules/locust/swarm.tf

```diff
@@ -41,6 +41,31 @@ resource "null_resource" "locust_swarm_session" {
         RETRY_COUNT=$(($RETRY_COUNT+1))
       done
 
+      # Wait for target service to become rechable from within the locust pod before starting the test
+      RETRIES=100
+      echo "Waiting for ${var.locust_target_host} to become ready and reachable from all the locust pods..."
+      PODS=$(kubectl get pods --selector=component=worker -n locust -o=json --request-timeout=5s | jq -er '.items[].metadata.name')
+      for POD in $PODS; do
+        echo "Pod $POD"
+        POD_READY=false
+        RETRY_COUNT=1
+        while [ "$RETRY_COUNT" -lt "$RETRIES" -a "$POD_READY" != "true" ]; do
+          CODE="$(kubectl exec -t -n locust --request-timeout=2s "$POD" -- /bin/sh -c "curl -s -o /dev/null -w '%{http_code}' --max-time 1 '${var.locust_target_host}/health'" 2>/dev/null)"
```

Comment:
I guess that this will trigger the "Pod Exec Check" alert. I can not think on a better way to check this, perhaps adding this feature to the locust image as part of the application workflow.

Reply:
Thanks Alf, yes, and I did not realize this. I think I'll move this inside the docker image.